core/chains/orm.go

```diff
@@ -0,0 +1,243 @@
+package chains
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/lib/pq"
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/core/services/pg"
+)
+
+// Chain is a generic DB chain for any configuration type.
+// CFG normally implements sql.Scanner and driver.Valuer, but that is not enforced here.
```

Comment:
I started out enforcing this, but it would require signature changes with cascading effects, since we pass around a value but the pointer implements the methods.

Reply:
Sort of, but less significantly. No need to impose any restrictions on future chain's ID type :shrug: